NPE in PathUtils

**Describe the bug**  
The method, isOsCaseSensitive(), attempts to create a file, and then has a finally block which attempts to delete said file, without checking if the File is null.

[11/13/25, 11:37:15:205 EST] 00000068 MBeanServerHe >  invoke Entry  
                                 WebSphere:name=com.ibm.websphere.config.mbeans.FeatureListMBean
                                 generate
                                 utf-8
                                 en
                                 null
                                 java.lang.String
                                 java.lang.String
                                 java.lang.String
                                 com.ibm.ws.jmx.connector.converter.JSONConverter@dbeb10f5
[11/13/25, 11:37:17:811 EST] 00000068 MBeanServerHe <  invoke Exit  
                                 {keySystemOut=java.lang.ExceptionInInitializerError
	at java.base/java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:175)
	at com.ibm.ws.kernel.feature.internal.generator.FeatureList.writeApiSpiJars(FeatureList.java:489)
	at com.ibm.ws.kernel.feature.internal.generator.FeatureList.writeFeature(FeatureList.java:286)
	at com.ibm.ws.kernel.feature.internal.generator.FeatureList.writeFeatureList(FeatureList.java:172)
	at com.ibm.ws.kernel.feature.internal.generator.FeatureListGenerator.generate(FeatureListGenerator.java:90)
	at com.ibm.ws.kernel.feature.internal.generator.FeatureListGenerator.main(FeatureListGenerator.java:42)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:586)
	at com.ibm.ws.kernel.boot.cmdline.UtilityMain.internal_main(UtilityMain.java:175)
	at com.ibm.ws.kernel.boot.cmdline.UtilityMain.main(UtilityMain.java:55)
	at com.ibm.ws.kernel.boot.cmdline.Main.main(Main.java:54)
Caused by: java.lang.NullPointerException: Cannot invoke "java.io.File.delete()" because "caseSensitiveFile" is null
	at com.ibm.wsspi.kernel.service.utils.PathUtils.isOsCaseSensitive(PathUtils.java:142)
	at com.ibm.wsspi.kernel.service.utils.PathUtils.<clinit>(PathUtils.java:64)
	... 10 more
, keyReturnCode=21}


**Steps to Reproduce**  
Problem seen while trying to setup a Liberty z/OS server with the adminCenter at 25.0.0.9.  They're unable to get the configuration displayed in the adminCenter and I've traced the failure back to a JMX request (from the adminCenter javascript) for:

 /mbeans/WebSphere%3Aname=com.ibm.websphere.config.mbeans.FeatureListMBean/operations/generate

This request returns a JSON response indicating an error:

{
    "value": {
        "keySystemOut": "java.lang.ExceptionInInitializerError\n\tat java.base/java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:175)\n\tat com.ibm.ws.kernel.feature.internal.generator.FeatureList.writeApiSpiJars(FeatureList.java:489)\n\tat com.ibm.ws.kernel.feature.internal.generator.FeatureList.writeFeature(FeatureList.java:286)\n\tat com.ibm.ws.kernel.feature.internal.generator.FeatureList.writeFeatureList(FeatureList.java:172)\n\tat com.ibm.ws.kernel.feature.internal.generator.FeatureListGenerator.generate(FeatureListGenerator.java:90)\n\tat com.ibm.ws.kernel.feature.internal.generator.FeatureListGenerator.main(FeatureListGenerator.java:42)\n\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:586)\n\tat com.ibm.ws.kernel.boot.cmdline.UtilityMain.internal_main(UtilityMain.java:175)\n\tat com.ibm.ws.kernel.boot.cmdline.UtilityMain.main(UtilityMain.java:55)\n\tat com.ibm.ws.kernel.boot.cmdline.Main.main(Main.java:54)\nCaused by: java.lang.NullPointerException: Cannot invoke \"java.io.File.delete()\" because \"caseSensitiveFile\" is null\n\tat com.ibm.wsspi.kernel.service.utils.PathUtils.isOsCaseSensitive(PathUtils.java:142)\n\tat com.ibm.wsspi.kernel.service.utils.PathUtils.<clinit>(PathUtils.java:64)\n\t... 10 more\n",
        "keyReturnCode": "21"

**Expected behavior**  
Configuration should be displayed

**Diagnostic information:**  
 - OpenLiberty Version: [e.g. 0 - 25.0.0.11]
 - Affected feature(s) adminCenter

--- a/dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/PathUtils.java
+++ b/dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/PathUtils.java
@@ -106,7 +106,7 @@ public class PathUtils {
                 if (caseSensitiveFile != null && (caseSensitiveFile.delete() || !caseSensitiveFile.exists())) {
                     try {
                         if (caseSensitiveFile.createNewFile()) {
-                            // We created "caseSensitive", so check if "CASeSENSITIVE" exists.
+                            // We created "caseSensitive", so check if "CASEsENSITIVE" exists.
                             // new File("A").equals(new File("a")) returns true on Windows, but
                             // OS/400 returns false even though the files are the same, so use
                             // getCanonicalFile() first, which allows the comparison to succeed.
@@ -120,7 +120,7 @@ public class PathUtils {
                     } catch (IOException ioe) {
                         // auto FFDC
                     } finally {
-                        if (caseSensitiveFile != null && caseSensitiveFile.exists()) {
+                        if (caseSensitiveFile != null) {
                             caseSensitiveFile.delete();
                         }
                     }
@@ -128,33 +128,30 @@ public class PathUtils {
             }
         }
 
+        // Fallback: test using temp file
         try {
             // Need to double check, since the above code is intended to be run in an
             // OSGi environment, not a Java SE / JUnit env
             caseSensitiveFile = File.createTempFile("caseSENSITIVEprefix", "TxT");
-            boolean iAmCaseSensitive = !getCanonicalFile(caseSensitiveFile).equals(new File(caseSensitiveFile.getAbsolutePath().toUpperCase()));
-            if (iAmCaseSensitive) {
-                return true;
-            }
+            // Compare canonical files to properly detect case sensitivity
+            boolean iAmCaseSensitive = !getCanonicalFile(caseSensitiveFile).equals(getCanonicalFile(new File(caseSensitiveFile.getAbsolutePath().toUpperCase())));
+            return iAmCaseSensitive;
         } catch (Exception e) {
             // We can't tell if this OS is case sensitive or not.
             // Assume we might not be case sensitive.
-            // Log the exception to help diagnose issues (e.g., temp directory permissions on z/OS)
+            // Generate FFDC to help diagnose issues (e.g., temp directory permissions, security restrictions)
             // Include diagnostic information about the temp directory configuration
             String tmpDir = System.getProperty("java.io.tmpdir", "unknown");
             String osName = System.getProperty("os.name", "unknown");
             String userName = System.getProperty("user.name", "unknown");
             FFDCFilter.processException(e, PathUtils.class.getName(), "isOsCaseSensitive",
-                new Object[] { "java.io.tmpdir=" + tmpDir, "os.name=" + osName, "user.name=" + userName });
+                                       new Object[] { "java.io.tmpdir=" + tmpDir, "os.name=" + osName, "user.name=" + userName });
             return false;
         } finally {
-            // Only delete if the file was successfully created
-            if (caseSensitiveFile != null && caseSensitiveFile.exists()) {
+            if (caseSensitiveFile != null) {
                 caseSensitiveFile.delete();
             }
         }
-        // Something went wrong. Assume we might be not be case sensitive.
-        return false;
     }
 
     /**

--- a/dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/PathUtils.java
+++ b/dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/PathUtils.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2011, 2017 IBM Corporation and others.
+ * Copyright (c) 2011, 2025 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at

--- a/dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/PathUtils.java
+++ b/dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/PathUtils.java
@@ -120,7 +120,9 @@ public class PathUtils {
                     } catch (IOException ioe) {
                         // auto FFDC
                     } finally {
-                        caseSensitiveFile.delete();
+                        if (caseSensitiveFile != null && caseSensitiveFile.exists()) {
+                            caseSensitiveFile.delete();
+                        }
                     }
                 }
             }
@@ -137,9 +139,19 @@ public class PathUtils {
         } catch (Exception e) {
             // We can't tell if this OS is case sensitive or not.
             // Assume we might not be case sensitive.
+            // Log the exception to help diagnose issues (e.g., temp directory permissions on z/OS)
+            // Include diagnostic information about the temp directory configuration
+            String tmpDir = System.getProperty("java.io.tmpdir", "unknown");
+            String osName = System.getProperty("os.name", "unknown");
+            String userName = System.getProperty("user.name", "unknown");
+            FFDCFilter.processException(e, PathUtils.class.getName(), "isOsCaseSensitive",
+                new Object[] { "java.io.tmpdir=" + tmpDir, "os.name=" + osName, "user.name=" + userName });
             return false;
         } finally {
-            caseSensitiveFile.delete();
+            // Only delete if the file was successfully created
+            if (caseSensitiveFile != null && caseSensitiveFile.exists()) {
+                caseSensitiveFile.delete();
+            }
         }
         // Something went wrong. Assume we might be not be case sensitive.
         return false;

